src/multivariate/dirichletmultinomial.jl

```diff
@@ -85,10 +85,26 @@ function suffstats{T<:Real}(::Type{DirichletMultinomial}, x::Matrix{T})
     d, m = size(x)
     s = zeros(d, n)
     for k in 1:n, i in 1:m, j in 1:d
-        @inbounds s[j, k] += Float64(x[j, i] >= k)
+        if x[j, i] >= k
+            @inbounds s[j, k] += 1.0
+        end
     end
     DirichletMultinomialStats(n, s, m)
 end
+function suffstats{T<:Real}(::Type{DirichletMultinomial}, x::Matrix{T}, w::Array{Float64})
+    length(w) == size(x, 2) || throw(ArgumentError("Inconsistent argument dimensions."))
+    n = sum(x[:, 1])
+    all(sum(x, 1) .== n) || error("Each sample in X should sum to the same value.")
+    d, m = size(x)
+    s = zeros(d, n)
+    tw = 0.0
+    for k in 1:n, i in 1:m, j in 1:d
```

Comment:
If you put the `@inbounds` here rather than at the assignment, you also get the benefit for the `getindex` right below this line. (Similar comments apply to other places in the diff.) That's why I had initially suggested `@inbounds for` specifically.